parquet-avro/src/main/java/org/apache/parquet/avro/AvroSchemaConverter.java

```diff
@@ -605,4 +611,14 @@ private static String namespace(String name, Map<String, Integer> names) {
     Integer nameCount = names.merge(name, 1, (oldValue, value) -> oldValue + 1);
     return nameCount > 1 ? name + nameCount : null;
   }
+
+  /* Avro <= 1.9 does not support conversions to LocalTimestamp{Micros, Millis} classes */
+  private static boolean avroVersionSupportsLocalTimestampTypes() {
```

Comment:
What do you think of mocking this method to capture the behavior of supporting and not-supporting local timestamps.

Reply:
I'll give it a try with Powermock!

I might start a thread on the mailing list this week discussing Avro version support more broadly, to formalize which Avro versions are currently supported/any planned EOL dates? I've been thinking it would also be good to have some automated testing of parquet-avro on different Avro versions... it would just be a little complicated to set up since we'd need either a separate Maven module per supported Avro version, or some very clever classloading :)

Reply:
Hey @clairemcginty, that's a good idea!

What do we consider the lower-bound? :D Avro 1.8.2 is May 2017, so that's pretty old, but I'm aware of some breaking API changes because we exposed some Jackson classes in the public API.

To do cross-testing, we can do something similar to we do with Hadoop 2, you can get some inspiration there: https://github.com/apache/parquet-java/blob/d4384d3f2e7703fab6363fd9cd80a001e9561db2/pom.xml#L638-L643

Reply:
I think 1.8 is a reasonable lower bound! for context on our use case, we use the [Beam](https://github.com/apache/beam) library, which until very recently was tightly coupled with 1.8. Things are now more flexible, but there's a lot of inertia surrounding 1.8 -- the breaking changes surrounding logical types make it a non-trivial upgrade 😅 

I'll take a look at the hadoop setup! Avro is a bit tricky because we need to both set the Avro-compiler version, and load the Avro core library, for whatever version of Avro we want to test...

Reply:
Thanks for the context, that helps. I see that 1.8.2 is [still in there](https://github.com/apache/beam/blob/b3a874f476646676bf7eb909cae738a696f36303/sdks/java/extensions/avro/build.gradle#L40), so that sounds reasonable to me 👍